src/components/Documents/DocumentsDesktop.jsx

```diff
@@ -0,0 +1,164 @@
+    { field: 'Type', minWidth: 120, flex: 1, headerAlign: 'center', align: 'center' },
+    { field: 'Description', minWidth: 120, flex: 1, headerAlign: 'center', align: 'center' },
+    { field: 'Upload Date', minWidth: 120, flex: 1, headerAlign: 'center', align: 'center' },
+    { field: 'Expiration Date', minWidth: 120, flex: 1, headerAlign: 'center', align: 'center' },
+    {
+      field: 'Preview',
+      minWidth: 100,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      sortable: false,
+      filterable: false,
+      renderCell: (data) => {
+        const [id, fileUrl] = data.value;
+        return (
+          <GridActionsCellItem
+            key={`Preview:${String(id)}`}
+            icon={<FileOpenIcon />}
+            onClick={() => handlers.onPreview(fileUrl)}
+            label="Preview"
+          />
+        );
+      }
+    },
+    {
+      field: 'Share',
+      minWidth: 100,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      sortable: false,
+      filterable: false,
+      renderCell: (data) => {
+        const [id, name, type] = data.value;
+        return (
+          <GridActionsCellItem
+            key={`Share:${String(id)}`}
+            icon={<ShareIcon />}
+            onClick={() => handlers.onShare('document', name, type)}
+            label="Share"
+          />
+        );
+      }
+    },
+    {
+      field: 'Delete',
+      minWidth: 100,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      sortable: false,
+      filterable: false,
+      renderCell: (data) => {
+        const [document] = data.value;
+        return (
+          <GridActionsCellItem
+            key={`Delete:${String(document.id)}`}
+            icon={<DeleteOutlineOutlinedIcon />}
+            onClick={() => handlers.onDelete(document)}
+            label="Delete"
+          />
+        );
+      }
+    }
+  ];
+
+  return (
+    <DataGrid
+      columns={columnTitlesArray}
+      rows={documents.map((document) => ({
+        id: document.id,
+        Name: document.name,
+        Type: getTypeText(document.type),
+        Description: document.description || '[Not Provided]',
+        'Upload Date': document?.uploadDate.toLocaleDateString(),
+        'Expiration Date': document?.endDate?.toLocaleDateString() || '[Not Provided]',
+        Preview: [document.id, document.fileUrl],
+        Share: [document.id, document.name, document.type],
+        Delete: [document]
+      }))}
+      pageSizeOptions={[10]}
+      initialState={{
+        pagination: {
+          paginationModel: { pageSize: 10, page: 0 }
+        }
+      }}
+      slots={{
+        toolbar: CustomToolbar
+      }}
+      sx={{
+        '.MuiDataGrid-columnHeader': {
+          background: theme.palette.primary.light,
+          color: 'white'
+        },
+        '.MuiDataGrid-columnSeparator': {
+          display: 'none'
+        }
+      }}
```

Comment:
Nitpick: I saw in other components in the changed files you had the styles defined outside the prop. Was thinking it'll be best if we should do it one way or the other like this in order to keep things consistent.

Reply:
You got it, should be set